### PR TITLE
More performance tuning on ValueSet expansion.

### PR DESCRIPTION
**Pull request by dmuylwyk**

### CODE REVIEW

**Comment by dmuylwyk** [comment]
Note to self: need to address bug when expanding with bad URL parameter instead of context.

**Comment by dmuylwyk** [comment]
Re: https://github.com/jamesagnew/hapi-fhir/pull/1468#issuecomment-529028294

This is fixed. There's still an issue where ValueSet.id is passed instead of ValueSet.url as the `url` parameter for the $expand operation. Legacy-mode expansion is used instead of persistence-backed expansion, and larger ValueSets may result in a timeout.

**Comment by dmuylwyk** [comment]
@jamesagnew, this is ready for review. Thanks!

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/term/BaseHapiTerminologySvcImpl.java [hunk 1]
```diff
@@ -527,58 +529,64 @@ private void populateExpansionComponent(ValueSet.ValueSetExpansionComponent theE
...
-		logConceptsExpanded(theTermValueSet, conceptsExpanded);
+		logDesignationsExpanded("Finished expanding designations. ", theTermValueSet, designationsExpanded);
+		logConceptsExpanded("Finished expanding concepts. ", theTermValueSet, conceptsExpanded);
 	}
 
-	private void logConceptsExpanded(TermValueSet theTermValueSet, int theConceptsExpanded) {
+	private void logConceptsExpanded(String thePrefix, TermValueSet theTermValueSet, int theConceptsExpanded) {
 		if (theConceptsExpanded > 0) {
-			ourLog.info("Have expanded {} concepts in ValueSet[{}]", theConceptsExpanded, theTermValueSet.getUrl());
+			ourLog.info("{}Have expanded {} concepts in ValueSet[{}]", thePrefix, theConceptsExpanded, theTermValueSet.getUrl());
```
Comment (jamesagnew) [inline hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/term/BaseHapiTerminologySvcImpl.java]: Is this right? If `thePrefix` is just intended to be a log explanation, maybe rename it to `theLogDescriptionPrefix` or something? Not a huge deal, but it looks.. odd...

Comment (dmuylwyk) [reply hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/term/BaseHapiTerminologySvcImpl.java]: Fixed.

**Pull request merged**